pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -9164,6 +9164,450 @@ spec:
+
+	tests := []struct {
+		name                string
+		trs                 []*v1beta1.TaskRun
+		prs                 []*v1beta1.PipelineRun
+		expectedPipelineRun *v1beta1.PipelineRun
+		expectedTaskRuns    []*v1beta1.TaskRun
+	}{{
+		name: "matrixed pipelinetask with retries, where one taskrun has failed and another one is running",
+		trs: []*v1beta1.TaskRun{
+			mustParseTaskRunWithObjectMeta(t,
+				taskRunObjectMeta("pr-platforms-and-browsers-0", "foo",
+					"pr", "p", "platforms-and-browsers", false),
+				`
+spec:
+  params:
+  - name: platform
+    value: linux
+  - name: browser
+    value: chrome
+  resources: {}
+  serviceAccountName: test-sa
+  taskRef:
+    name: mytask
+  timeout: 1h0m0s
+  retries: 1
+status:
+  conditions:
+  - type: Succeeded
+    status: "False"
+`),
+			mustParseTaskRunWithObjectMeta(t,
+				taskRunObjectMeta("pr-platforms-and-browsers-1", "foo",
+					"pr", "p", "platforms-and-browsers", false),
+				`
+spec:
+  params:
+  - name: platform
+    value: mac
+  - name: browser
+    value: chrome
+  resources: {}
+  serviceAccountName: test-sa
+  taskRef:
+    name: mytask
+  timeout: 1h0m0s
+  retries: 1
+status:
+  conditions:
+  - type: Succeeded
+    status: "Unknown"
+`),
+		},
+		prs: []*v1beta1.PipelineRun{
+			parse.MustParsePipelineRun(t, `
+metadata:
+  name: pr
+  namespace: foo
+  annotations: {}
+  labels:
+    tekton.dev/pipeline: p
+spec:
+  serviceAccountName: test-sa 
+  pipelineRef:
+    name: p
+status:
+  pipelineSpec:
+    tasks:
+    - name: platforms-and-browsers
+      retries: 1
+      taskRef:
+        name: mytask
+        kind: Task
+      matrix:
+        - name: platform
+          value:
+            - linux
+            - mac
+      params:
+        - name: browser
+          value: chrome
+  conditions:
+  - type: Succeeded
+    status: "Unknown"
+    reason: "Running"
+    message: "Tasks Completed: 0 (Failed: 0, Cancelled 0), Incomplete: 1, Skipped: 0"
+  childReferences:
+  - apiVersion: tekton.dev/v1beta1
+    kind: TaskRun
+    name: pr-platforms-and-browsers-0
+    pipelineTaskName: platforms-and-browsers
+  - apiVersion: tekton.dev/v1beta1
+    kind: TaskRun
+    name: pr-platforms-and-browsers-1
+    pipelineTaskName: platforms-and-browsers
+  taskRuns: {}
+  runs: {}
+`),
+		},
+		expectedPipelineRun: parse.MustParsePipelineRun(t, `
+metadata:
+  name: pr
+  namespace: foo
+  annotations: {}
+  labels:
+    tekton.dev/pipeline: p
+spec:
+  serviceAccountName: test-sa 
+  pipelineRef:
+    name: p
+status:
+  pipelineSpec:
+    tasks:
+    - name: platforms-and-browsers
+      retries: 1
+      taskRef:
+        name: mytask
+        kind: Task
+      matrix:
+        - name: platform
+          value:
+            - linux
+            - mac
+      params:
+        - name: browser
+          value: chrome
+  conditions:
+  - type: Succeeded
+    status: "Unknown"
+    reason: "Running"
+    message: "Tasks Completed: 0 (Failed: 0, Cancelled 0), Incomplete: 1, Skipped: 0"
```

Comment:
why do we expect 1 incomplete taskrun? shouldn't it be 2?

Reply:
this message is about "tasks", not "taskruns" -- there's only one matrixed task